org.eclipse.acute/src/org/eclipse/acute/dotnetnew/DotnetNewWizardPage.java

```diff
@@ -189,27 +191,38 @@ public void createControl(Composite parent) {
 		Label projectTemplateLabel = new Label(container, SWT.NONE);
 		projectTemplateLabel.setText("Project Template");
 
-		templatesMap = DotnetNewAccessor.getTemplates();
-
 		List list = new List(container, SWT.V_SCROLL);
-		GridData listBoxData = new GridData(SWT.LEFT, SWT.CENTER, true, false, 2, 1);
+		GridData listBoxData = new GridData(SWT.FILL, SWT.CENTER, true, false, 2, 1);
 		list.setLayoutData(listBoxData);
-
 		templateViewer = new ListViewer(list);
 		templateViewer.setContentProvider(new ArrayContentProvider());
-		if (!templatesMap.isEmpty()) {
-			listBoxData.heightHint = 100;
-			SortedSet<String> templatesSortedSet = new TreeSet<>();
-			for (String temp : templatesMap.keySet()) {
-				templatesSortedSet.add(temp);
-			}
-			for (String temp : templatesSortedSet) {
-				templateViewer.add(temp);
-			}
-		} else {
-			templateViewer.add("No available templates");
-			list.setEnabled(false);
-		}
+		templateViewer.add("Loading templates");
+		templateViewer.getList().setEnabled(false);
+
+		new Thread(() -> {
```

Comment:
In Eclipse world, we try to avoid directly using Thread which are complex to handle. Instead, we prefer to use `Job`s which is a bit similar but with a grain of control that's better for the IDE.